compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -829,7 +825,7 @@ static void build_enum_enumerate_function(EnumType* et) {
   fn->insertFormalAtTail(arg);
   fn->where = new BlockStmt(new CallExpr("==", arg, et->symbol));
 
-  et->symbol->defPoint->insertAfter(new DefExpr(fn));
+  baseModule->block->insertAtTail(new DefExpr(fn));
```

Comment:
Since you made this comment, I moved `==`, `!=` and `=` on enums out into module code.  This left the enumeration routine as the only one of the compiler-generated routines that was being inserted at the enum's scope rather than the global scope, so this improves consistency.  That said, I hear what you are saying about making methods, and perhaps standard operators, on types available when that type is available, but that's not something I'm going to be able to fix in this PR.